sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceBuilderTest.java

```diff
@@ -85,6 +86,51 @@ public void testNewFromResourceSpecRefined() {
         }
     }
 
+    /*
+        name: "cpus"
+        type: SCALAR
+        scalar {
+          value: 1.0
+        }
+        reservations {
+          role: "base-role"
+          type: STATIC
+        }
+        reservations {
+          principal: "test-principal"
+          labels {
+            labels {
+              key: "resource_id"
+              value: "a395f14b-3cc8-4009-9dc4-51838b423aed"
+            }
+          }
+          role: "test-role"
+          type: DYNAMIC
+        }
+    */
+    @Test
+    public void testRefineStaticResource() {
+        ResourceRefinementCapabilityContext context = new ResourceRefinementCapabilityContext(Capabilities.getInstance());
+        try {
+            ResourceSpec resourceSpec = new DefaultResourceSpec(
+                    "cpus",
+                    value,
+                    TestConstants.ROLE,
+                    TestConstants.PRE_RESERVED_ROLE,
+                    TestConstants.PRINCIPAL,
+                    "CPUS_ENV_KEY");
+            ResourceBuilder resourceBuilder = ResourceBuilder.fromSpec(resourceSpec, Optional.empty());
+
+            Protos.Resource resource = resourceBuilder.build();
+            Assert.assertEquals(2, resource.getReservationsCount());
+            validateScalarResourceRefined(TestConstants.PRE_RESERVED_ROLE, resource);
```

Comment:
Actually change to this helper was unneeded, thanks for mentioning it.  I think I understand your point, but will have to think about this.  The point of the helper is to validate that a certain format is maintained across many tests.  I guess you're saying a single test should validate this, and then things should be structured that we need not worry that in some other situation something else could occur, as code structure would make such a scenario impossible.  I'll think about it.